### PR TITLE
Add versioned documentation

### PR DESCRIPTION
**Pull request by ChethanUK**
To show relevant documentation to the users for the current version and hide old versions docs and also hide unreleased documentation from users.

### CODE REVIEW

File: website2/website/versioned_docs/version-0.20.1/guides-ui-guide.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
+---
+id: version-0.20.1-guides-ui-guide
```
Comment (joshfischer1108) [inline website2/website/versioned_docs/version-0.20.1/guides-ui-guide.md]: Is this `0.20.1` version page supposed to be in here?

**Comment by joshfischer1108** [comment]
The `make python-docs` is still failing on my machine.  Output is below.. Will look into it later.  ```
...left out for brevity ...
Writing heronpy-0.0.0/setup.cfg
creating dist
Creating tar archive
removing 'heronpy-0.0.0' (and everything under it)
Requirements: ['protobuf==3.4.0', 'six']
Target //scripts/packages:pypkgs failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 19.315s, Critical Path: 16.09s
INFO: 23 processes: 23 local.
FAILED: Build did NOT complete successfully
make: *** [python-docs] Error 1
```

File: website2/website/versioned_docs/version-0.20.1/guides-ui-guide.md (hunk 1 above)
Comment (ChethanUK) [reply website2/website/versioned_docs/version-0.20.1/guides-ui-guide.md]: Yeah.

**Comment by joshfischer1108** [comment]
I did some more digging it seems an error is being thrown during python setup on my mac.

```
Execution platform: @bazel_tools//platforms:host_platform
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: invalid command 'bdist_wheel'
/var/folders/f3/3m4jtv_s3z1fwlcg911rgwjw0000gn/T/heronpy.XXXXX.4irPZomO
```

Which I believe references the line of code [here](https://github.com/ChethanUK/incubator-heron/blob/e5a9ecb1a3f3ea3b4d211c13aecbdeda47a1d9af/scripts/packages/BUILD#L653) in the `genrule` called `generate-heronpy-package`.  The CI build passes because this path is not included in the build.  @ChethanUK I think you said this worked on Ubuntu correct?  Could you give me the details of the vm you are running?

**Comment by ChethanUK** [comment]
> Could you give me the details of the vm you are running?

Any Ubuntu will be fine. I ran tests on `ubuntu-minimal-1804-bionic-v20190628`

**Comment by joshfischer1108** [comment]
@nwangtw could you look at this if you have some time.  @ChethanUK has been helping a lot with the website. Here we are trying to add a set of versioned docs along with the generated pydocs .  I can't get the pydoc command to run successfully on my local mac. Could you try?

**Comment by joshfischer1108** [comment]
@ChethanUK Could you resolve the conflicts on this branch please?

File: website2/website/Makefile [hunk 2]
```diff
@@ -6,3 +6,6 @@ javadocs:
 
 python-docs:
 	@scripts/python-doc-gen.sh 0.0.0
+
+buildsite:
+    @scripts/build-site.sh
```
Comment (nwangtw) [inline website2/website/Makefile]: use TAB instead.

File: website2/website/scripts/build-site.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
```
Comment (nwangtw) [inline website2/website/scripts/build-site.sh]: change file permission to be executable.

**Comment by nwangtw** [comment]
Nice. "yarn start" works for me~ . And then I can open the website in browser by localhost:3000

I need to make two small changes to make it work on my mac. Please see the inline comments.

Anything else I need to verify?

**Comment by joshfischer1108** [comment]
@nwangtw I was having trouble generating pydocs via the make command in the file.  Were you able execute this step?

**Comment by nwangtw** [comment]
> @nwangtw I was having trouble generating pydocs via the make command in the file. Were you able execute this step? I had some issues with the pip install it looks. See the comments of mine above. If it works for you then I need to check my python installation.

Yeah. I believe it is part of "make buildsite" which worked ok to me. Also in command line:

$ pydoc
pydoc - the Python documentation tool

pydoc <name> ...
    Show text documentation on something.  <name> may be the name of a
    Python keyword, topic, function, module, or package, or a dotted
....

I didn't need to install anything, so looks like part of Python installation. My Python version is 2.7.10 just fyi.

**Comment by joshfischer1108** [comment]
> > @nwangtw I was having trouble generating pydocs via the make command in the file. Were you able execute this step? I had some issues with the pip install it looks. See the comments of mine above. If it works for you then I need to check my python installation.
> 
> Yeah. I believe it is part of "make buildsite" which worked ok to me. Also in command line:
> 
> $ pydoc
> pydoc - the Python documentation tool
> 
> pydoc ...
> Show text documentation on something. may be the name of a
> Python keyword, topic, function, module, or package, or a dotted
> ....
> 
> I didn't need to install anything, so looks like part of Python installation. My Python version is 2.7.10 just fyi.

I have the same version..  Admittedly, I rarely use python.  Looks like I need to do some fixing of my own environment.

File: website2/docs/getting-started-local-single-node.md [hunk 4]
```diff
@@ -20,7 +20,7 @@ sidebar_label: Local (Single Node)
     under the License.
 -->
 
-> The current version of Heron is **{{% heronVersion %}}**
+> The current version of Heron is {{heron:version}}
```
Comment (nwangtw) [inline website2/docs/getting-started-local-single-node.md]: when I open the webpage locally, the line is "The current version of Heron is {{heron:version}}". The heron:version is not resolved. Same thing happens to pyversion I think.

Is this expected?

Comment (ChethanUK) [reply website2/docs/getting-started-local-single-node.md]: Ya, we discussed offline to fix the versions like heron versions and bazel versions in docs after this PR is done...

Comment (nwangtw) [reply website2/docs/getting-started-local-single-node.md]: Sounds good to me~ Let me approve and merge it. Thanks.

**Pull request merged**